src/test/java/com/shapesecurity/salvation/PolicyQueryingTest.java

```diff
@@ -484,6 +509,155 @@
         assertFalse(p.allowsScriptFromSource(URI.parse("wss://example.com/PATH")));
         assertFalse(p.allowsScriptFromSource(new GUID("data:")));
         assertFalse(p.allowsScriptFromSource(new GUID("custom.scheme:")));
+        
+        p = Parser.parse("font-src *", "http://example.com");
+        assertTrue(p.allowsFontFromSource(URI.parse("http://example.com")));
+        assertTrue(p.allowsFontFromSource(URI.parse("https://example.com")));
+        assertTrue(p.allowsFontFromSource(URI.parse("http://example.com:81")));
+        assertTrue(p.allowsFontFromSource(URI.parse("ftp://example.com")));
+        assertTrue(p.allowsFontFromSource(URI.parse("ftp://example.com:80")));
+        assertTrue(p.allowsFontFromSource(URI.parse("http://example.com/path")));
+        assertTrue(p.allowsFontFromSource(URI.parse("http://example.com/PATH")));
+        assertTrue(p.allowsFontFromSource(URI.parse("ws://example.com/PATH")));
+        assertTrue(p.allowsFontFromSource(URI.parse("wss://example.com/PATH")));
+        assertFalse(p.allowsFontFromSource(new GUID("data:")));
+        assertFalse(p.allowsFontFromSource(new GUID("custom.scheme:")));
+        
+        p = Parser.parse("font-src http://*", "http://example.com");
+        assertTrue(p.allowsFontFromSource(URI.parse("http://example.com")));
+        assertFalse(p.allowsFontFromSource(URI.parse("https://example.com")));
```

Comment:
This is supposed to be true in CSP3. But it's unrelated to your PR; I will open an issue.